### PR TITLE
Recreates the @when decorator for a new age.

### LINKED ISSUE #2: When-When Merger

### PR DESCRIPTION
**Pull request by flosincapite**

### CODE REVIEW

**Comment by flosincapite** [comment]
I still need to write unit tests for both of the files touched here.

Test plan:
- update requirements in `cronenbroguelike` to point here;
- use `@whimsylib.when.when` (we should do some `__init__` magic to make that nicer) in place of `@adventurelib.when`;
- see how it plays out.

**Comment by xmunoz** [comment]
This needs to be merged first: https://github.com/Cronenbrogues/cronenbroguelike/pull/122

**Comment by flosincapite** [comment]
This still has some problems; I am working on it.

**Comment by xmunoz** [comment]
> This still has some problems; I am working on it.

Ok, I'll wait until this is merged before trying to remove adventurelib references from cronenbroguelike.

**Comment by flosincapite** [comment]
@abjose IT DOES THIS WHENEVER IT'S TOLD

File: whimsylib/when.py [hunk 1]
```diff
@@ -1,92 +1,126 @@
...
+        return self._prefix
 
-def when(command, context=None, **kwargs):
-    """Decorator for command functions."""
+    def match(self, command):
+        return self._pattern.match(command)
+
+
+def _no_command_matches(command):
+    print(f'I don\'t understand "{command}."')
```
Comment (xmunoz) [inline whimsylib/when.py]: Blasphemer!

**Review by xmunoz — CHANGES_REQUESTED** [review]

**Review by xmunoz — APPROVED** [review]
It's lit

**Pull request merged**